packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -268,8 +286,9 @@ class _CheckboxState extends State<Checkbox> with TickerProviderStateMixin {
           return _CheckboxRenderObjectWidget(
             value: widget.value,
             tristate: widget.tristate,
-            activeColor: widget.activeColor ?? themeData!.toggleableActiveColor,
+            activeColor: effectiveActiveColor,
             checkColor: widget.checkColor ?? const Color(0xFFFFFFFF),
+            useActiveColorInDisabledState: activeColor is MaterialStateProperty<Color>,
             inactiveColor: enabled ? themeData!.unselectedWidgetColor : themeData!.disabledColor,
             focusColor: widget.focusColor ?? themeData.focusColor,
```

Comment:
Instead of adding `useActiveColorInDisabledState`, can't you just update the `inactiveColor` when it's not enabled? Maybe I am missing something here.
```dart
final inactiveColor;
if (enabled) {
  inactiveColor = themeData!.unselectedWidgetColor;
else {
  inactiveColor = activeColor is MaterialStateProperty<Color> ? effectiveActiveColor : themeData!.disabledColor;
}
```

I assume those lerp functions wouldn't be used anyways since it can't change state when disabled?

Reply:
Yeah I feel like you can do something like this and avoid adding the new bool

Reply:
We discussed this offline, I definitely would prefer do something along these lines because I don't prefer the bool I added. I'm only hesitant because using `activeColor` in the `inactive` state would be a little misleading. If it was called "backgroundColor", I would be more comfortable applying it in the inactive state, and overriding `inactiveColor`. 

For now I think I will leave my solution as is as it is the simplest change to the public facing API at the expense of some plumbing in the implementation.